### DISCUSSION TITLE
FileUpload field - how to populate with previously saved data?

### DISCUSSION THREAD

**Discussion by studiowizjo** [filamentphp/filament#724] [Help] [answered]
Hello,
I have a form in which I have multi upload field (tried MultipleFileUpload and SpatieMediaLibraryMultipleFileUpload) and they works perfectly. Data is saved in database, but the problem is that when I reload page, I would like this fields to be prepopulated with previously added photos. 

I couldnt find the way to do it (both in Forms documentation and Media Library by Spatie). My model is binded to form and for any other fields data is prepopulated, but not for images.

For now, all data from form is saved in one "body" field as json in array, but it doesn`t have to be this way. I there is a need, I can treat photos different way and store it on separate field or even in separate table (eg. "media" table from Spatie Media Library).

Here is my component simplified code:

```php
use InteractsWithForms, WithMedia;

public $model = \App\Models\Product::class;

public $editing = null;

public function mount(): void
{
	$this->editing
		? $this->loadDataForEditing()
		: $this->loadDataForCreating();

	$this->form->fill($this?->editing?->toArray());
}

public function loadDataForEditing(): void
{
	$this->editing = $this->model::find($this->editing);
}

public function loadDataForCreating(): void
{
	$this->editing = $this->makeBlankModel();
	$this->saveButtonLabel = 'Dodaj';
	$this->formTitle = 'Dodaj';
}

public function makeBlankModel()
{
	return $this->model::make([
		'name' => null,
		'price' => null,
		'slug' => null,
		'body' => [],
		'created_at' => now(),
	]);
}

protected function getFormModel()
{
	return $this->model;
}

public function save()
{
	$this->editing->fill($this->form->getState());
	$this->editing->save();

	$this->dispatchBrowserEvent('notify', 'Zapisano');
}
```

and my Product.php model:

```php
class Product extends Model implements HasMedia
{
	use HasFactory, InteractsWithMedia;

	public $fillable = [
		'name',
		'slug',
		'price',
		'body',
	];

	public $casts = [
		'body' => 'json',
	];
}
```

**Comment by danharrin** [comment, marked as answer]
For the edit form, you need to pass a model *instance* to `getFormModel()` and not just the class name, otherwise the component doesn't know where to look for images :)

Also, when you don't have a saved model instance available to the form until after it's submitted, remember to call `saveRelationships()` after the form has saved with the new model instance.

I'd do something like this:

```php

protected function getFormModel()
{
	return $this->editing?->exists() ? $this->editing : $this->model;
}

public function save()
{
	$this->editing->fill($this->form->getState());
	$this->editing->save();
        $this->form->model($this->editing)->saveRelationships();

	$this->dispatchBrowserEvent('notify', 'Zapisano');
}
```

**Reply by Roxeen12** [reply]
In which file should be this code? I have the same issue but I don't know where to put this code. Thanks!

**Reply by danharrin** [reply]
The original user was using the form builder, are you using the form builder or the admin panel?

**Reply by Roxeen12** [reply]
Form panel

**Reply by Roxeen12** [reply]
I have tried to use the save() function but I occur an error
"Method 'App\Post::save()' is not compatible with method 'Illuminate\Database\Eloquent\Model::save()'."
